book/src/proposals/one-round-finality.md

```diff
@@ -0,0 +1,116 @@
+# Practical Finality in One Round of Consensus
+
+The cluster can achieve a high degree of finality though Tower
+consensus as the cluster commits to an exponentially increasing of
+safety for a block.  This safety is based on the commitment of any
+voting validator from switching votes to an alternative fork.  It
+is possible to achieve finality a different way, leaders can agree
+not to propose alternative forks.
+
+If only a part of the cluster has voted, then the set of all the
+leaders that haven't voted and the leader schedule can be used to
+simulate how lockout will grow in the future.  This calculation is
+based on the assumption that the partition is static, and that the
+same leaders will continue voting together.
+
+## Overview
+
+Leaders commit to producing a block that chains to the leaders last
+vote. Breaking this commitment is slashable.  The leader is free
+to switch forks and vote for an alternative chain.
+
+Once a block has been voted on, the subsequent leader schedule will
+reflect the likelihood of the block being finalized.
+
+For example:
+* 80% of the cluster voted for the block
+
+* The schedule for the next 5 leaders looks like:
+
+    * V,V,V, V,V,V,V, N,N,N,N, V,V,V,V, V,V,V,V
+
+Where V indicates that the leader voted on the block and N indicates
+that they haven't. By the time the `N` leader is reached, its likely
+this block would be voted on 7 times, and the lockout for this block
+would prevent the `N` leader from proposing an alternative fork
+which doesn't contain this block.
+
+## Calculating the Probability of a block succeeding
+
+This calculation assumes that the voting nodes are on the same
+partition and will vote with the same response rate.
+
+The likelihood of the block being finalized will approach the
+probability of voting streaks that accumulate enough lockout to
+skip over all the non-voting leaders.
+
+In the above example if the probability of any block being dropped
+is 2%, then probability of a streak of size 3 in the next 7 blocks
+is 98.07%, and the probability of a streak of size 4 is 97.7%, which
+should roughly equal the probability of the fork still being live
+after the non-voting leader produced an alternative fork.  Looking
+at the leader schedule ahead of the block, the probability of the
+block surviving is at least equal to the probability of continuous
+run of streaks that keep doubling the lockout.
+
+Each streak needs to produce a lockout that is 2x longer than the
+lockout from the previous streak.  Given that 20% of leaders are
```

Comment:
Here it might be better to say 20% of the stake (proportional to the number of slots in the leader schedule).